### PR TITLE
Support more special chars in camera passwords

### PR DESCRIPTION
**Pull request by NateMeyer**
For issue #4257. Match all non-whitespace characters in password field.  Need to explicitly match `://` at the beginning so it isn't considered the user delimiter.

### CODE REVIEW

**Comment by NateMeyer** [comment]
Looks like go2rtc still has an issue with this.

**Comment by NickM-27** [comment]
> Looks like go2rtc still has an issue with this.

What issue are you seeing with go2rtc?

File: frigate/test/test_camera_pw.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 class TestUserPassCleanup(unittest.TestCase):
     def setUp(self) -> None:
         self.rtsp_with_pass = "rtsp://user:password@192.168.0.2:554/live"
-        self.rtsp_with_special_pass = "rtsp://user:password#$@@192.168.0.2:554/live"
+        self.rtsp_with_special_pass = "rtsp://user:password`~!@#$%^&*()-_;',.<>:\"\{\}\[\]@192.168.0.2:554/live"
```
Comment (NickM-27) [inline frigate/test/test_camera_pw.py:11]: I'd like to keep the last character of the password as `@` in the test so we can be sure it won't interfere with the required @ in the rtsp url

**Comment by NateMeyer** [comment]
> > Looks like go2rtc still has an issue with this.
> 
> What issue are you seeing with go2rtc?

I think we need to use the `escape_special_characters` in restream.py when passing ffmpeg input paths to go2rtc

File: frigate/test/test_camera_pw.py (hunk 1 above)
Comment (NateMeyer) [reply frigate/test/test_camera_pw.py]: I updated the test to have another `@` at the end of the password.

**Review by NickM-27 — APPROVED** [review]

**Pull request merged**